File: app/src/main/java/com/divercity/app/features/home/HomeActivity.java
package com.divercity.app.features.home;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.graphics.Typeface;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.res.ResourcesCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.View;

import com.divercity.app.R;
import com.divercity.app.databinding.ActivityHomeBinding;
import com.divercity.app.features.home.groups.GroupsFragment;

import javax.inject.Inject;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;

public class HomeActivity extends AppCompatActivity implements HasSupportFragmentInjector, GroupsFragment.OnFragmentInteractionListener {

    @Inject
    DispatchingAndroidInjector<Fragment> supportFragmentInjector;

    ActivityHomeBinding binding;
    NavController navController;

    public static Intent getCallingIntent(Context context){
        return new Intent(context, HomeActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_home);
        navController = Navigation.findNavController(HomeActivity.this, R.id.nav_host_fragment);
        NavigationUI.setupWithNavController(binding.bottomNavigationView, navController);
        setupBottomNavigationView();
    }

    private void setupBottomNavigationView() {
        Typeface typeface = ResourcesCompat.getFont(this, R.font.avenir_medium);
        binding.bottomNavigationView.setTypeface(typeface);
        binding.bottomNavigationView.enableShiftingMode(false);
        binding.bottomNavigationView.setItemIconTintList(null);

//        Glide
//                .with(this)
//                .load("https://apinew.pincapp.com/images/default_avatar.png")
//                .into(new SimpleTarget<Drawable>() {
//                    @Override
//                    public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
//                        StateListDrawable states = new StateListDrawable();
//                        states.addState(new int[]{android.R.attr.state_pressed}, resource);
//                        states.addState(new int[]{android.R.attr.state_checked}, resource);
//                        states.addState(new int[]{}, resource);
//                        Menu menu = binding.bottomNavigationView.getMenu();
//                        MenuItem te = menu.findItem(R.id.notificationsFragment);
//                        te.setIcon(states);
//                        binding.imgTest.setImageDrawable(resource);
//                    }
//                });

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
//        MenuInflater inflater = getMenuInflater();
//        inflater.inflate(R.menu.menu_toolbar, menu);
//
//        SearchView searchView = (SearchView) menu.findItem(R.id.action_search).getActionView();
//        searchView.setIconifiedByDefault(false);
//        searchView.setQueryHint(getString(R.string.search));
        return true;
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }

    public void showBottomNavigationView() {
        binding.bottomNavigationView.setVisibility(View.VISIBLE);
    }

    public void hideBottomNavigationView() {
        binding.bottomNavigationView.setVisibility(View.GONE);
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return supportFragmentInjector;
    }

}

File: app/src/main/java/com/divercity/app/features/home/HomeActivity.kt
package com.divercity.app.features.home

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.graphics.drawable.StateListDrawable
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v4.content.res.ResourcesCompat
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.SimpleTarget
import com.bumptech.glide.request.transition.Transition
import com.divercity.app.R
import com.divercity.app.features.dialogs.CompletedProfileDialogFragment
import com.divercity.app.features.groups.TabGroupsFragment
import com.divercity.app.features.home.home.HomeFragment
import com.divercity.app.features.home.notifications.NotificationsFragment
import com.divercity.app.features.jobs.TabJobsFragment
import com.divercity.app.features.profile.ProfileFragment
import dagger.android.support.DaggerAppCompatActivity
import kotlinx.android.synthetic.main.activity_home.*
import kotlinx.android.synthetic.main.view_toolbar.view.*
import javax.inject.Inject

class HomeActivity : DaggerAppCompatActivity(){

    @Inject
    lateinit var viewModelFactory: ViewModelProvider.Factory

    lateinit var viewModel: HomeActivityViewModel

    companion object {
        private const val INTENT_EXTRA_PARAM_SHOW_DIALOG_PROFILE = "showDialogProfile"

        fun getCallingIntent(context: Context, showDialogProfileCompleted: Boolean): Intent {
            val intent = Intent(context, HomeActivity::class.java)
            intent.putExtra(INTENT_EXTRA_PARAM_SHOW_DIALOG_PROFILE, showDialogProfileCompleted)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(HomeActivityViewModel::class.java)
        setContentView(R.layout.activity_home)
        showDialogProfileComplete(intent.getBooleanExtra(INTENT_EXTRA_PARAM_SHOW_DIALOG_PROFILE, false))
        setupBottomNavigationView()
        savedInstanceState ?: selectItem(0)
        setSupportActionBar(include_toolbar.toolbar)
    }

    /**
     * Fragment creation according to tab position
     */
    fun selectItem(position: Int) {
        var selectedFragment: Fragment = HomeFragment.newInstance()
        when (position) {
            0 -> selectedFragment = HomeFragment.newInstance()
            1 -> selectedFragment = TabGroupsFragment.newInstance()
            2 -> selectedFragment = TabJobsFragment.newInstance()
            3 -> selectedFragment = NotificationsFragment.newInstance()
            4 -> selectedFragment = ProfileFragment.newInstance()
        }
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.container, selectedFragment)
        transaction.commit()
    }

    private val myOnNavigationItemSelectedListener =
            BottomNavigationView.OnNavigationItemSelectedListener { item ->
                when (item.itemId) {
                    R.id.menu_item_home -> selectItem(0)
                    R.id.menu_item_groups -> selectItem(1)
                    R.id.menu_item_jobs -> selectItem(2)
                    R.id.menu_item_activity -> selectItem(3)
                    R.id.menu_item_profile -> selectItem(4)
                }
                return@OnNavigationItemSelectedListener true
            }

    fun showDialogProfileComplete(boolean: Boolean) {
        if (boolean) {
            val dialog = CompletedProfileDialogFragment.newInstance()
            dialog.show(supportFragmentManager, null)
        }
    }

    private fun setupBottomNavigationView() {

        bottom_nav_view.apply {
            val typeface = ResourcesCompat.getFont(this@HomeActivity, R.font.avenir_medium)
            setTypeface(typeface)
            isItemHorizontalTranslationEnabled = false
            itemIconTintList = null

            // To fix bug that large labels dissapears on menu
            for (i in 0 until itemCount) {
                val view = getLargeLabelAt(i)
                view.setPadding(0, 0, 0, 0)
            }

            onNavigationItemSelectedListener = myOnNavigationItemSelectedListener
        }

        Glide
                .with(this)
                .load(viewModel.getProfilePictureUrl())
                .apply(RequestOptions().circleCrop())
                .into(object : SimpleTarget<Drawable>() {
                    override fun onResourceReady(
                            resource: Drawable,
                            transition: Transition<in Drawable>?
                    ) {
                        val states = StateListDrawable()
                        states.addState(intArrayOf(android.R.attr.state_pressed), resource)
                        states.addState(intArrayOf(android.R.attr.state_checked), resource)
                        states.addState(intArrayOf(), resource)
                        val menu = bottom_nav_view.menu
                        val te = menu.findItem(R.id.menu_item_profile)
                        te.icon = states
                    }
                })
    }

}
